### PR TITLE
Add unit tests for Subscriber Persistence Manager classes

### PR DESCRIPTION
**Pull request by UdeshAthukorala**
### Proposed changes in this pull request

$Subject

### CODE REVIEW

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridSubscriberPersistenceManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        assertEquals(decryptedSubscriberFromStorage.getPropertyDTO(SUBSCRIBER_PASSWORD_KEY).getValue(),
+                SAMPLE_SUBSCRIBER_PASSWORD_1);
+    }
+
+    @Test(priority = 6)
+    public void testGetSubscriberInRegistry() throws Exception {
+
+        registrySubscriberPersistenceManager.addSubscriber(sampleHolder1);
+
+        PublisherDataHolder subscriberFromDb =
```
Comment (darshanasbg) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridSubscriberPersistenceManagerTest.java:83]: misleading variablename

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridSubscriberPersistenceManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        assertEquals(subscriberFromDb.getPropertyDTO(SUBSCRIBER_USERNAME_KEY).getValue(),
+                sampleHolder1.getPropertyDTO(SUBSCRIBER_USERNAME_KEY).getValue());
+        assertEquals(subscriberFromDb.getPropertyDTO(SUBSCRIBER_PASSWORD_KEY).getValue(),
+                SAMPLE_ENCRYPTED_PASSWORD1);
+
+        // Retrieve the subscriber with the decrypted secrets.
+        PublisherDataHolder decryptedSubscriberFromStorage =
+                subscriberPersistenceManager.getSubscriber(SAMPLE_SUBSCRIBER_ID_1, true);
+        assertEquals(decryptedSubscriberFromStorage.getPropertyDTO(SUBSCRIBER_PASSWORD_KEY).getValue(),
+                SAMPLE_SUBSCRIBER_PASSWORD_1);
```
Comment (darshanasbg) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridSubscriberPersistenceManagerTest.java:75]: This part is duplicated in two tests `testGetSubscriberInDatabase()` and `testGetSubscriberInRegistry()`, right? Shall we extract this to a method then?

Similar behavior is there in test methods as well.

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridSubscriberPersistenceManagerTest.java (hunk 1 above)
Comment (UdeshAthukorala) [reply components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridSubscriberPersistenceManagerTest.java]: Addressed with https://github.com/wso2/carbon-identity-framework/compare/3c296c23619d23b55ab1d555c12a582d7e16c508..762117e1906d5e7d3f44439db38b0b99dcb3315e

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridSubscriberPersistenceManagerTest.java (hunk 2 above)
Comment (UdeshAthukorala) [reply components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridSubscriberPersistenceManagerTest.java]: Addressed with https://github.com/wso2/carbon-identity-framework/compare/3c296c23619d23b55ab1d555c12a582d7e16c508..762117e1906d5e7d3f44439db38b0b99dcb3315e

**Review by darshanasbg — APPROVED** [review]

**Pull request merged**